src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -870,9 +934,12 @@ void GoogleCalendarSyncAdaptor::purgeDataForOldAccount(int oldId, SocialNetworkS
 void GoogleCalendarSyncAdaptor::beginSync(int accountId, const QString &accessToken)
 {
     SOCIALD_LOG_DEBUG("beginning Calendar sync for Google, account" << accountId);
-    bool needCleanSync = !wasLastSyncSuccessful(accountId);
+    bool needCleanSync = false;
+    bool lastSyncSuccessful = wasLastSyncSuccessful(accountId, &needCleanSync);
     if (needCleanSync) {
-        SOCIALD_LOG_INFO("last sync was not successful; performing clean sync");
+        SOCIALD_LOG_INFO("performing clean sync");
+    } else if (!lastSyncSuccessful) {
+        SOCIALD_LOG_INFO("last sync was not successful, attempting to recover without clean sync");
```

Comment:
I wanted to have the possibility to set this value explicitly, so that we can set it in the future for:

a) error handling in some case which we currently have not considered

b) some UI setting to allow the user to perform a clean sync just in case things go pear shaped

Currently it's not used just because I can't see a case which would be completely unrecoverable using the current code, but that doesn't mean that such an unrecoverable situation does not exist.  And I'd prefer not to rewrite the .ini handling code again, if/when we hit that situation in the future.